### PR TITLE
[BugFix] Fix be crash because of dropping tablet

### LINKED ISSUE #15957: Be crash because of dropping tablet
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. DO SCHEMA CHANGE

### Expected behavior (Required)
run success

### Real behavior (Required)
img_v2_2a5770f4-71c0-42e1-9cdc-8c3c51fccddg

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/15957

## Problem Summary(Required) ：

When we try to drop tablet(by manual or after schema change), we will try to close expired rowsets before clear tablet meta if tablet is primary key table. And if expired rowsets are still been reference by other threads, close will fail and be will crash.

Anyway, if expired rowsets are still been referenced by other threads when we try to drop tablet, crash is not necessary and output some error log is enough, the expired rowsets will be close after release.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by sevev** [comment]
run starrocks_be_unittest

File: be/src/storage/olap_common.h [hunk 1]
```diff
@@ -305,6 +305,8 @@ struct RowsetId {
         }
     }
 
+    int64_t id() { return hi & LOW_56_BITS; }
```
Comment (chaoyli) [inline be/src/storage/olap_common.h:308]: Why not set the state of tablet? The dropped tablet should wait the rowset been de-refereneced.

Comment (sevev) [reply be/src/storage/olap_common.h:308]: > Why not set the state of tablet? The dropped tablet should wait the rowset been de-refereneced.

The state of tablet is already set to `SHUTDOWN` before remove the rowset.

**Comment by sevev** [comment]
run all

File: be/src/storage/tablet_updates.cpp [hunk 2]
```diff
@@ -2971,6 +2981,18 @@ void TabletUpdates::_remove_unused_rowsets() {
     }
 }
 
+Status TabletUpdates::check_and_remove_rowset() {
+    std::lock_guard l1(_lock);
```
Comment (decster) [inline be/src/storage/tablet_updates.cpp:2985]: why adding lock?

Comment (sevev) [reply be/src/storage/tablet_updates.cpp]: > why adding lock?

The lock is to protect the remove operation. But it seems the lock is not necessary, it has been removed.

**Review by decster — DISMISSED** [review]

File: be/src/storage/tablet_updates.cpp [hunk 3]
```diff
@@ -2971,6 +2981,17 @@ void TabletUpdates::_remove_unused_rowsets() {
     }
 }
 
+Status TabletUpdates::check_and_remove_rowset() {
+    _remove_unused_rowsets(true);
+    if (!_unused_rowsets.empty()) {
+        std::string msg =
+                strings::Substitute("some rowsets of tablet: $0 are still been referenced", _tablet.tablet_id());
+        LOG(WARNING) << "msg";
```
Comment (chaoyli) [inline be/src/storage/tablet_updates.cpp:2989]:
```suggestion
        LOG(WARNING) << msg;

Comment (sevev) [reply be/src/storage/tablet_updates.cpp]: > 

done, changed in https://github.com/StarRocks/starrocks/pull/15959/commits/a8c877d82a733a59318f35f294a3980c41f1b4ac

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
@mergify backport branch-2.4 branch-2.3

**Pull request merged**